### DISCUSSION TITLE
Using prisma but unable to perform user sign up below is the error I am facing

### DISCUSSION THREAD

**Discussion by Revaycolizer** [prisma/prisma#18872] [Q&A]
### Bug description

error - PrismaClientInitializationError:
Invalid `prisma.user.create()` invocation:

Error in connector: Error creating a database connection. (An error occurred dur
ing DNS resolution: proto error: io error: A socket operation was attempted to a
n unreachable host. (os error 10065))
    at pn.handleRequestError (D:\project\Luxee-Hub\node_modules\@prisma\client\r
untime\library.js:176:6750)
    at pn.handleAndLogRequestError (D:\project\Luxee-Hub\node_modules\@prisma\cl
ient\runtime\library.js:176:5907)
    at D:\project\Luxee-Hub\node_modules\@prisma\client\runtime\library.js:179:3
087
    at async D:\project\Luxee-Hub\node_modules\@prisma\client\runtime\library.js
:179:3294
    at async t._executeRequest (D:\project\Luxee-Hub\node_modules\@prisma\client
\runtime\library.js:179:10755)
    at async t._request (D:\project\Luxee-Hub\node_modules\@prisma\client\runtim
e\library.js:179:10484)
    at async POST (webpack-internal:///(sc_server)/./app/api/register/route.ts:1
7:18)
    at async AppRouteRouteHandler.execute (D:\project\Luxee-Hub\node_modules\nex
t\dist\server\future\route-handlers\app-route-route-handler.js:330:26)
    at async AppRouteRouteHandler.handle (D:\project\Luxee-Hub\node_modules\next
\dist\server\future\route-handlers\app-route-route-handler.js:410:30)
    at async RouteHandlerManager.handle (D:\project\Luxee-Hub\node_modules\next\
dist\server\future\route-handler-managers\route-handler-manager.js:16:24)
    at async doRender (D:\project\Luxee-Hub\node_modules\next\dist\server\base-s
erver.js:744:32)
    at async cacheEntry1.responseCache.get.incrementalCache.incrementalCache (D:
    at async POST (webpack-internal:///(sc_server)/./app/api/register/route.ts:1
7:18)
    at async AppRouteRouteHandler.execute (D:\project\Luxee-Hub\node_modules\nex
t\dist\server\future\route-handlers\app-route-route-handler.js:330:26)
    at async AppRouteRouteHandler.handle (D:\project\Luxee-Hub\node_modules\next
\dist\server\future\route-handlers\app-route-route-handler.js:410:30)
    at async RouteHandlerManager.handle (D:\project\Luxee-Hub\node_modules\next\
dist\server\future\route-handler-managers\route-handler-manager.js:16:24)
    at async doRender (D:\project\Luxee-Hub\node_modules\next\dist\server\base-s
erver.js:744:32)
    at async cacheEntry1.responseCache.get.incrementalCache.incrementalCache (D:
\project\Luxee-Hub\node_modules\next\dist\server\base-server.js:931:28)
    at async D:\project\Luxee-Hub\node_modules\next\dist\server\response-cache\i
ndex.js:83:36 {
  clientVersion: '4.13.0',
  errorCode: undefined
}

### Prisma information

```prisma
// Add your schema.prisma
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User{
  id String @id @default(auto()) @map("_id")  @db.ObjectId
  name String?
  username String? @unique
  bio String?
  email String? @unique
  emailVerified DateTime?
  image String?
  coverImage String?
  profileImage String?
  hashedPassword String?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  followingIds String[] @db.ObjectId
  hasNotification Boolean?

  posts Post[]
  comments Comment[]
  notifications Notification[]

}

model Post{
  id String @id @default(auto()) @map("_id")  @db.ObjectId
  body String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  userId String @db.ObjectId
  likedIds String[] @db.ObjectId
  video String[] @db.ObjectId

  user User @relation(fields: [userId], references: [id], onDelete:Cascade)

  comments Comment[]
}

model Comment{
  id String @id @default(auto()) @map("_id")  @db.ObjectId
  body String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  userId String @db.ObjectId
  postId String @db.ObjectId

  user User @relation(fields: [userId], references: [id],onDelete:Cascade)
  post Post @relation(fields: [postId], references: [id],onDelete:Cascade)
}

model Notification{
  id String @id @default(auto()) @map("_id")  @db.ObjectId
  body String
  userId String @db.ObjectId
  createdAt DateTime @default(now())

  user User @relation(fields: [userId], references: [id],onDelete:Cascade)
}
```ts
// Add your code using Prisma Client
```
import { PrismaClient } from "@prisma/client";

declare global{
    var prisma : PrismaClient | undefined
}

const client = globalThis.prisma || new PrismaClient()
if(process.env.NODE_ENV !== 'production') globalThis.prisma = client;

export default client;

```


### Environment & setup

- OS: <!--[e.g. macOS, Windows, Debian, CentOS, ...]-->
- Database: <!--[PostgreSQL, MySQL, PlanetScale, MariaDB, SQL Server, SQLite, MongoDB or CockroachDB]-->
- Node.js version: <!--[Run `node -v` to see your Node.js version]-->
MongoDB

### Prisma Version

```
Windows 10
prisma: 4.13.0
Nodejs: 16.17.0
MongoDB Atlas
```

**Comment by nurul3101** [comment]
Hello @Revaycolizer 👋 

Are you able to connect to your MongoDB Atlas database instance through a database IDE like MongoDB Compass?

**Reply by Revaycolizer** [reply]
I was able to just send create schema only and at first I created one user successfully then I wasn't able to perform registration

**Reply by nurul3101** [reply]
I didn't understand what you meant by 

> able to just send create schema only

Can you please elaborate?

**Reply by Revaycolizer** [reply]
Only I could create a model and push it to my database but I couldn't connect to it to perform user registration

**Reply by nurul3101** [reply]
Okay, so you were able to push the models to the database, but were unable to create database records?

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my last message. Let us know in case you need any assistance.

**Comment by Kkura02** [comment]
Hello @Revaycolizer , I have a question. Are you working on this project behind a corporate proxy?
